subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/DefaultTaskOutputFilesRepository.java

```diff
@@ -73,13 +73,13 @@ public boolean preVisitDirectory(DirectorySnapshot directorySnapshot) {
                 }
 
                 @Override
-                public void visit(PhysicalSnapshot fileSnapshot) {
+                public void visit(FileSystemLocationSnapshot fileSnapshot) {
```

Comment:
I left those in intentionally, since `visit` is called only for file snapshots: missing files and regular files. Do you think it would be better to use `snapshot` as parameter name, too?

Reply:
I see, this is the actual method declaration: https://github.com/gradle/gradle/blob/1c9f846b3885f4e599e2fd5a2e41f0253b40e9d9/subprojects/core/src/main/java/org/gradle/internal/snapshot/PhysicalSnapshotVisitor.java#L31-L34

So if this can only be called for regular files and missing stuff, then we should break this up into two methods, one receiving `FileSnapshot` the other `MissingSnapshot`.